app/enterprise/0.33-x/developer-portal/best-practice.md

```diff
@@ -0,0 +1,204 @@
+```
+admin_listen=off
+portal=on
+```
+We will also want to ensure all other instances of Kong have the portal boolean set to off. This ensures your portal is only being served by a single Kong instance.
+
+## <a name=css></a>Customizing The Portal Design with CSS
+
+The Developer Portal has two files that contain CSS and are used to control the design. The main CSS file is `unauthenticated/theme-css` which contains the bulk of the CSS styles for the default layout. The second file is `unauthenticated/custom-css` where owners add and edit their own custom css. Both of these files can be accessed by non authenticated pages. You can access the custom-css file by using the Admin API<br />
+`curl http://localhost:8001/files/unauthenticated/custom-css`<br /><br />
+
+This will return the a JSON object that represents the custom-css file. It contains metadata about the file, and the `contents` field contains the actual CSS data. You should save the `contents` to a file, and edit it to your liking. Edit the CSS to customize the look of your portal.
+
+When you have customized the CSS, you will then upload it to the Files API. This will make the Dev Portal use the CSS for when it serves content, and show your custom design. <br />
+`curl -X PATCH http://localhost:8001/files/unauthenticated/custom-css -F contents=@custom.css`<br /><br />
+
+In this example the name of the file on disk is `custom.css`. You are free to name this whatever you like, so long as you send your changes to the `unauthenticated/custom-css` file. This specific place is where the Kong Dev Portal will look for custom CSS.
+
+
+## <a name=js></a>Customizing Front End Behavior with JavaScript
+
+Including custom javascript is similar to customizing CSS.
+
+In order to use custom javascript we need to download the `custom-js` file, make our custom changes, and then upload those changes to the `custom-js` file.
+
+Use this command to get the JSON object that represents the `custom-js` file<br />
+`curl http://localhost:8001/files/custom-js`<br /><br />
+
+Once you have this, save the value of the `contents` field to a file. This is the file we will use to add custom javascript. Once the file has been edited it will need to be uploaded to the `custom-js` file. Use this command to upload your changes. This command assumes the name of the file is `custom-js`<br />
+`curl -X PATCH http://localhost:8001/files/custom-js -F contents=@custom-js`<br /><br />
+
+Once uploaded your changes will be served on each page request.
+
+
+## <a name=git></a>Using Source Control with Git
+
+Kong gives you the ability to edit files directly in the Kong GUI. While this may be easy, it will not maintain any changes. Each file will only exist in it's most current form. We have found the best approach is to have a dedicated repo for you portal.
```

Comment:
"...it will not maintain any changes" is unclear, and needs to be better explained.